File: src/main/java/com/itheima/activiti/service/PendingNotificationService.java
package com.itheima.activiti.service;

import com.itheima.activiti.entity.PendingNotification;
import java.util.List;
import java.util.Map;

/**
 * 待发送通知服务接口
 */
public interface PendingNotificationService {
    
    // 保存待发送通知
    PendingNotification savePendingNotification(PendingNotification pendingNotification);
    
    // 更新待发送通知
    PendingNotification updatePendingNotification(PendingNotification pendingNotification);
    
    // 根据ID查询待发送通知
    PendingNotification getPendingNotificationById(String id);
    
    // 根据条件查询待发送通知列表
    List<PendingNotification> getPendingNotificationsByParams(Map<String, Object> params);
    
    // 查询待发送的通知（状态为pending且下次重试时间<=当前时间）
    List<PendingNotification> getPendingNotifications();
    
    // 根据状态查询待发送通知
    List<PendingNotification> getPendingNotificationsByStatus(String status);
    
    // 根据类型查询待发送通知
    List<PendingNotification> getPendingNotificationsByType(String type);
    
    // 删除待发送通知
    boolean deletePendingNotification(String id);
    
    // 批量删除待发送通知
    boolean deletePendingNotifications(List<String> ids);
    
    // 处理待发送通知
    boolean processPendingNotification(String id);
    
    // 批量处理待发送通知
    int processPendingNotifications();
}

File: frontend/src/services/PendingNotificationService.js
import axios from '../utils/axiosConfig';

class PendingNotificationService {
  // 获取待发送通知列表
  async getPendingNotifications(params = {}) {
    try {
      const response = await axios.get('/notification/pending', { params });
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error('获取待发送通知列表失败:', error);
      throw error;
    }
  }

  // 获取待发送通知详情
  async getPendingNotification(id) {
    try {
      const response = await axios.get(`/notification/pending/${id}`);
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error(`获取待发送通知ID: ${id} 详情失败:`, error);
      throw error;
    }
  }

  // 根据状态获取待发送通知
  async getPendingNotificationsByStatus(status) {
    try {
      const response = await axios.get(`/notification/pending/status/${status}`);
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error(`获取状态为${status}的待发送通知失败:`, error);
      throw error;
    }
  }

  // 根据类型获取待发送通知
  async getPendingNotificationsByType(type) {
    try {
      const response = await axios.get(`/notification/pending/type/${type}`);
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error(`获取类型为${type}的待发送通知失败:`, error);
      throw error;
    }
  }

  // 处理待发送通知
  async processPendingNotification(id) {
    try {
      const response = await axios.post(`/notification/pending/process/${id}`);
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error(`处理待发送通知ID: ${id} 失败:`, error);
      throw error;
    }
  }

  // 批量处理待发送通知
  async processPendingNotifications() {
    try {
      const response = await axios.post('/notification/pending/process/batch');
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error('批量处理待发送通知失败:', error);
      throw error;
    }
  }

  // 删除待发送通知
  async deletePendingNotification(id) {
    try {
      const response = await axios.delete(`/notification/pending/${id}`);
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error(`删除待发送通知ID: ${id} 失败:`, error);
      throw error;
    }
  }

  // 批量删除待发送通知
  async deletePendingNotifications(ids) {
    try {
      const response = await axios.delete('/notification/pending/batch', { data: ids });
      // 由于axios响应拦截器已经处理了response.data，所以直接返回response
      return response;
    } catch (error) {
      console.error('批量删除待发送通知失败:', error);
      throw error;
    }
  }
}

export default new PendingNotificationService();